clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -447,10 +552,36 @@ public ConfigDef withClientSaslSupport() {
             String joined = Utils.join(undefinedConfigKeys, ",");
             throw new ConfigException("Some configurations in are referred in the dependents, but not defined: " + joined);
         }
-        // parse all known keys
+        // Handle all known ConfigKey objects.
         Map<String, Object> values = new HashMap<>();
-        for (ConfigKey key : configKeys.values())
-            values.put(key.name, parseValue(key, props.get(key.name), props.containsKey(key.name)));
+        for (ConfigKey key : configKeys.values()) {
+            if (values.containsKey(key.name))
```

Comment:
Worth adding a comment explaining why this is needed. Also a comment in `configKeys` explaining that duplicate values may exist would be helpful.